cmd/dp-search-query/main.go

```diff
@@ -36,23 +36,27 @@ func main() {
 		os.Exit(1)
 	}
 
+	gracefulShutdown := func() {
+		log.Info("Commencing graceful shutdown", log.Data{"graceful_shutdown_timeout": cfg.GracefulShutdownTimeout})
+		ctx, cancel := context.WithTimeout(context.Background(), cfg.GracefulShutdownTimeout)
+
+		// stop any incoming requests before closing any outbound connections
+		api.Close(ctx)
+
+		log.Info("shutdown complete", nil)
+		cancel()
```

Comment:
Im not sure you need to call this? I think you can just call ctx.Done? I could be wrong though as its been a while